api/src/pcapi/routes/backoffice_v3/finance/blueprint.py

```diff
@@ -91,17 +93,23 @@ def get_finance_incident_cancellation_form(finance_incident_id: int) -> utils.Ba
     )
 
 
-def _cancel_finance_incident(incident: finance_models.FinanceIncident) -> None:
+def _cancel_finance_incident(incident: finance_models.FinanceIncident, comment: str) -> None:
     if incident.status == finance_models.IncidentStatus.CANCELLED:
         raise finance_exceptions.FinanceIncidentAlreadyCancelled
     if incident.status == finance_models.IncidentStatus.VALIDATED:
         raise finance_exceptions.FinanceIncidentAlreadyValidated
 
     incident.status = finance_models.IncidentStatus.CANCELLED
-    # Todo (akarki): log in history the cancellation
-
     repository.save(incident)
 
+    history_api.log_action(
+        history_models.ActionType.FINANCE_INCIDENT_CANCELLED,
+        author=current_user,
+        finance_incident=incident,
+        comment=comment,
+        save=True,
+    )
+
```

Comment:
On n'est pas à l'abri d'un échec d'écriture dans la base de données, donc dans ce cas pour garder les données cohérentes, il est mieux d'avoir les deux objets enregistrés dans la même transaction. Ainsi, on ne pourra pas avoir l'incident sauvegardé sans l'action associée. Ce qui donne : 
```suggestion
    action = history_api.log_action(
        history_models.ActionType.FINANCE_INCIDENT_CANCELLED,
        author=current_user,
        finance_incident=incident,
        comment=comment,
        save=False,
    )

    repository.save(incident, action)
```

Reply:
Ah bien vu :+1: